test/e2e/autoscaling/horizontal_pod_autoscaling_behavior.go

```diff
@@ -212,6 +204,139 @@ var _ = SIGDescribe("[Feature:HPA] [Serial] [Slow] Horizontal pod autoscaling (n
 			replicas := rc.GetReplicas()
 			framework.ExpectEqual(replicas == initPods, true, "had %s replicas, still have %s replicas after time deadline", initPods, replicas)
 		})
+
 	})
 
+	ginkgo.Describe("with scale limited by number of Pods rate", func() {
+		ginkgo.It("should scale up no more than given number of Pods per minute", func() {
+			ginkgo.By("setting up resource consumer and HPA")
+			initPods := 1
+			initCPUUsageTotal := initPods * usageForSingleReplica
+			limitWindowLength := 1 * time.Minute
+			podsLimitPerMinute := 1
+
+			rc := e2eautoscaling.NewDynamicResourceConsumer(
+				hpaName, f.Namespace.Name, e2eautoscaling.KindDeployment, initPods,
+				initCPUUsageTotal, 0, 0, int64(podCPURequest), 200,
+				f.ClientSet, f.ScalesGetter, e2eautoscaling.Disable, e2eautoscaling.Idle,
+			)
+			defer rc.CleanUp()
+
+			hpa := e2eautoscaling.CreateCPUHorizontalPodAutoscalerWithBehavior(
+				rc, int32(targetCPUUtilizationPercent), 1, 10,
+				e2eautoscaling.HPABehaviorWithScaleLimitedByNumberOfPods(true, false, int32(podsLimitPerMinute), int32(limitWindowLength.Seconds())),
+			)
+			defer e2eautoscaling.DeleteHPAWithBehavior(rc, hpa.Name)
+
+			ginkgo.By("triggering scale up by increasing consumption")
+			rc.ConsumeCPU(4 * usageForSingleReplica)
+
+			ginkgo.By("assuring no more than 1 replica was added during 1 minute")
+			rc.EnsureDesiredReplicasInRange(1, 2, limitWindowLength, hpaName)
+
+			ginkgo.By("assuring no more than 2 replicas were added during 2 minutes")
+			rc.EnsureDesiredReplicasInRange(1, 3, limitWindowLength, hpaName)
+
+			ginkgo.By("assuring no more than 3 replicas were added during 3 minutes")
```

Comment:
IMO 1 and 2 minutes checks are enough safety, let's save test run duration by removing the third minute check (also in scale down test).

Reply:
As discussed offline: refactored.